components/global/global_header.tsx

```diff
@@ -41,27 +26,49 @@ const GlobalHeader = () => {
         return null;
     }
 
+    const theme = {
+        ...lightTheme,
+        noStyleReset: true,
+        noFontFaces: true,
+        noDefaultStyle: true,
+        background: {
+            ...lightTheme.background,
+            shape: 'var(--sidebar-teambar-bg)',
+        },
+    };
+
+    // adding the dark theme for now to test how the colors "perform"
     return (
-        <HeaderContainer>
-            <ProductSwitcher/>
-            <AppSpectificContent>
-                {currentProductID !== null &&
-                    <Pluggable
-                        pluggableName={'Product'}
-                        subComponentName={'headerComponent'}
-                        pluggableId={currentProductID}
+        <>
+            <ThemeProvider theme={theme}>
+                <Grid<PShape>
```

Comment:
Sry to butt in here, but this is something I came up with, sooo... 

The Grid component is a "convenience" component. Leaving it alone it would add a new layer (div) around, but when using the element prop with another component it renders this component and just adds the grid stuff on top.

That's also the reason for the generic. The props for the injected component are not inferred from it (correctly). It is possible to use it without it, but typescript will complain about missing props, so I added it to prevent that.

I could have also used <T extends {}>, but that would kind of add a wildcard to the props and that I wanted to prevent as well.